lib/child_process.js

```diff
@@ -613,8 +594,41 @@ function normalizeSpawnArguments(file, args, options) {
 
 function spawn(file, args, options) {
   const child = new ChildProcess();
-
   options = normalizeSpawnArguments(file, args, options);
+
+  if (options.signal) {
+    const signal = options.signal;
+    // Validate signal, if present
+    validateAbortSignal(signal, 'options.signal');
+
+    const onAbortListener = onAbortSignal(child);
+
+    // Do nothing and throw if already aborted
+    if (signal.aborted) {
+      onAbortListener();
+    } else {
+      const config = { once: true };
+      const remove = () => {
+        signal.removeEventListener('abort', onAbortListener, config);
+      };
+
+      signal.addEventListener('abort', onAbortListener, config);
+      child.once('close', remove);
+    }
+
+    function onAbortSignal(childProcess) {
+      return function abortListener() {
+        process.nextTick(() => {
+          if (childProcess && childProcess.kill) {
+            childProcess.kill(options.killSignal);
+          }
+
+          childProcess.emit('error', new AbortError());
+        });
+      };
+    }
```

Comment:
This function is only used once, right? If so, I think we should inline it.

Reply:
I was trying to reduce the closure by injecting the `child` process instance. Are you suggesting to delete it and pass the attach to the closure again?

Reply:
I'm talking about `onAbortSignal`. I can see it used at line 604 and that's all. since it's just a wrapper around another function, let's remove it.

```suggestion
```